### PR TITLE
Added Qt6 support

### PR DESCRIPTION
**Pull request by mnesarco**
I have tested this with my own Qt6 App. I think it is a good initial support for Qt6.

### CODE REVIEW

File: src/main.cpp
Comment (TheAssassin) [inline src/main.cpp:73]: Why not just ask `qmake`? Call it with `-v`, it shows you the Qt version in use. It's easier than having the user specify this.

Comment (mnesarco) [reply src/main.cpp:73]: I will check, qmake -v output needs to be parsed to extract the major version.

Example qmake -v output:

```
➜  ~ qmake -v
QMake version 2.01a
Using Qt version 4.8.7 in /usr/lib/x86_64-linux-gnu

```

Comment (mnesarco) [reply src/main.cpp:73]: I will use `qmake -query QT_VERSION` it is perfect for the use case:

```
➜  ~ qmake -query QT_VERSION
4.8.7
➜  ~ Qt/5.15.0/gcc_64/bin/qmake -query QT_VERSION
5.15.0
➜  ~ qt-6.0.0/bin/qmake -query QT_VERSION
6.0.0
```

Comment (mnesarco) [reply src/main.cpp:73]: QT_VERSION is already available in `qmakeVars` so it will use that.

Comment (TheAssassin) [reply src/main.cpp:73]: Oh right... I should've remembered that, I guess. Well, it makes it a lot simpler, I guess.

Comment (mnesarco) [reply src/main.cpp]: it worked like a charm

**Comment by mnesarco** [comment]
@TheAssassin , I have updated it to autodetect qt version.

File: src/deployers/PluginsDeployerFactory.cpp
Comment (TheAssassin) [inline src/deployers/PluginsDeployerFactory.cpp:46]: I'd rather not instantiate the basic deployer manually, but would put the check in the top `if` clause. That way, it'll fall back to this one as well.

```cpp
    if (moduleName == "network" && qtMajorVersion < 6) {
```

File: src/deployers/PluginsDeployerFactory.h
Comment (TheAssassin) [inline src/deployers/PluginsDeployerFactory.h:26]: I think we should not use a default value. You keep track of the number anyway where the deployers are called.

File: src/main.cpp
Comment (TheAssassin) [inline src/main.cpp:108]: `atoi` is unsafe. Use `strtol`, please, and implement the necessary checks. Or maybe just use C++ streams.

File: src/main.cpp
Comment (TheAssassin) [inline src/main.cpp:175]: Please, no plain pointers. You could use a reference here as well. Or at least a smart pointer. Ideally, this decision is not taken here anyway. A separate function would be cleaner, as it gives you greater freedom when writing such code.

```cpp
const std::vector<QtModule>& getQtModules() {
    if (qtMajorVersion == 5) {
        return Qt5Modules;
    } else if (qtMajorVersion == &) {
        return Qt6Modules;
    }

    throw some_exception;
}

[...]

const std::vector<QtModule>& qtModules = getQtModules();
```

**Review by TheAssassin — CHANGES_REQUESTED** [review]
Looks mostly good. I'm not sure we had to copy the entire modules list, but on the other hand, it doesn't hurt either.

File: src/main.cpp
Comment (mnesarco) [reply src/main.cpp]: Ok, changes applied.

File: src/deployers/PluginsDeployerFactory.cpp
Comment (mnesarco) [reply src/deployers/PluginsDeployerFactory.cpp]: Applied

File: src/deployers/PluginsDeployerFactory.h
Comment (mnesarco) [reply src/deployers/PluginsDeployerFactory.h]: Applied

File: src/main.cpp
Comment (mnesarco) [reply src/main.cpp]: Applied

File: src/deployers/PluginsDeployerFactory.h
Comment (TheAssassin) [inline src/deployers/PluginsDeployerFactory.h:50]: You seem to have missed this occurrence.

File: src/qt-modules.h
Comment (TheAssassin) [inline src/qt-modules.h:179]: I think this is clearly an error, and should be treated accordingly. We don't support anything but Qt 5 and Qt 6.

I'd suggest something like

```suggestion
    throw std::runtime_error("Unknown Qt version: " + std::string(version));
```

Note: you need to include `stdexcept` and `string`.

Please add a blank line at the end of this file.

Comment (mnesarco) [reply src/qt-modules.h]: ok, changed.

File: src/deployers/PluginsDeployerFactory.h
Comment (mnesarco) [reply src/deployers/PluginsDeployerFactory.h]: ok, changed.

**Comment by mnesarco** [comment]
HI @TheAssassin , do you think the PR is ready now?

**Comment by spiderkeys** [comment]
Just wanted to add a note that I've successfully tested this out with my app as well, and all is working just fine.

Thanks for the effort, @mnesarco!

**Comment by TheAssassin** [comment]
Please don't push maintainers. I have many projects that want attention. Your PR is on the list.

File: src/main.cpp
Comment (TheAssassin) [inline src/main.cpp:175]: Code style: We use camelCase, so this should be `qtModules`.

**Review by TheAssassin — CHANGES_REQUESTED** [review]
One last minor issue, after this it's good to merge.

File: src/main.cpp
Comment (mnesarco) [reply src/main.cpp]: Ok, applied.

**Comment by mnesarco** [comment]
The failing test on CentOS7 x86_64 is not related with the code:

```
ERROR: Could not find plugin: appimage

```

**Comment by TheAssassin** [comment]
I'll handle that. Thanks a lot for your PR.

**Comment by TheAssassin** [comment]
I just spent a little over 30 minutes trying to implement this a little bit more "clean", by implementing an interface for the factories, as well as a factory for the factories. Quite some effort. I think I could do better with templates, but I'm too tired. Let's merge this for now, and improve on it later. This parameter isn't really a clean solution, but it's good enough for now.

**Pull request merged**
